src/main/java/iluvus/backend/api/controller/CommunityController.java

```diff
@@ -37,4 +38,14 @@ public ResponseEntity<String> createCommunity(@RequestBody Map<String, String> d
     public ResponseEntity<ArrayList<String>> getAllCommunity() {
         return ResponseEntity.ok().body(communityService.getAllCommunity());
     }
+
+    @PostMapping(value = "/join", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> joinCommunity(@RequestParam String userId, @RequestParam String communityId) {
```

Comment:
Right here, we do POST method, front end will send you something like:
```json
{
    "userId": "65b37131a41eb30c12349955",
    "communityId": "65b373c5715a405d31dd0c1a"
}
```
So the `@RequestParam` will not work, it is only for GET method.
I suggest to change to:
```java
@PostMapping(value = "/join", produces = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<String> joinCommunity(@RequestBody Map<String, String> data) {
        try {
            String userId = data.get("userId");
            String communityId = data.get("communityId");
//the rest of your code
```